platform/android/java/lib/src/org/godotengine/godot/Godot.java

```diff
@@ -221,34 +221,41 @@ private void onVideoInit() {
 		setContentView(layout);
 
 		// GodotEditText layout
-		GodotEditText edittext = new GodotEditText(this);
-		edittext.setLayoutParams(new ViewGroup.LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.WRAP_CONTENT));
+		GodotEditText editText = new GodotEditText(this);
+		editText.setLayoutParams(new ViewGroup.LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.WRAP_CONTENT));
 		// ...add to FrameLayout
-		layout.addView(edittext);
+		layout.addView(editText);
 
-		mView = new GodotView(this, xrMode, use_32_bits, use_debug_opengl);
-		layout.addView(mView, new LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT));
-		edittext.setView(mView);
-		io.setEdit(edittext);
+		GodotLib.setup(command_line);
```

Comment:
Yes that's the reason.

I can do it by command line instead as you suggested, but in this case we'll have less options when we want to handle the fallback to OpenGL when Vulkan is not supported on the device (at this point I'm not sure if we'll do that in c++ or java).

This part of the code is supposed to handle the case where `Main::setup` is called on the UI thread and it seems to work fine so far:
https://github.com/godotengine/godot/blob/989a223c5a6d6ba5b1b098be8983888cb49b2525/platform/android/java_godot_lib_jni.cpp#L197-L199

What part of the setup do you think is going to cause problems when not run on the render thread?

Reply:
My main worry is regarding thread safety since `setup` initializes some data structures that `setup2` and the rest of the logic access at runtime. 
With your changes, `setup` concludes before the render thread is started, so data access should be safe, especially given that Godot's data structures should be thread-safe.

@akien-mga @reduz It would be good to have someone familiar with Godot's C/C++ thread safety validates these changes as well.

Reply:
Ah nevermind, this will probably need to be changed and probably done again in setup2.
https://github.com/godotengine/godot/blob/master/main/main.cpp#L404

Reply:
> Ah nevermind, this will probably need to be changed and probably done again in setup2.
> https://github.com/godotengine/godot/blob/master/main/main.cpp#L404

This is already done in https://github.com/godotengine/godot/blob/master/platform/android/java_godot_lib_jni.cpp#L201.

Reply:
> I think there should not be a problem calling setup2() on a different thread as setup(), if setup2() will later become the rendering thread. That said, eventually we need to look at some point at optionally supporting a dedicated rendering thread on Android, given this works on Godot already in the rest of the platforms.

@reduz The thread that `setup2()` is invoked on is the dedicated rendering thread. Or were you referring to something else?

Reply:
oh no, it's the main thread for Godot. On android there is no render thread support, so you should assume both render and main should be the same for now.

I'm sure it could eventually be done, but I would get everything to work first.